helpers/README.md

```diff
@@ -0,0 +1,353 @@
+		.dlist = DL_PONYTAIL,
+		.scale = {
+			0.0055f,
+			0.0055f,
+			0.0055f
+		},
+		.segID = 0x0B,
+		.noDraw = true,
+	},
+	.spheres = {
+		.num = 6,
+		.centers = NULL,
+		.radius = 4.0f
+	},
+	.constraint = {
+		.lockRoot = false,
+		.rotStepCalc = {
+			.x = 45.0f,
+			.y = 45.0f,
+		},
+		.rotStepDraw = {
+			.y = 25.0f,
+			.x = 0.0f,
+		},
+	},
+	.rigidity = {
+		.num = 5,
+		.push = {
+			0,
+			0.2f,
+			-3.0f
+		},
+		.mult = 0.2f,
+		.rot = {
+			.x = 50.0f,
+			.y = 0.0f,
+			.z = 0.0f,
+		},
+	},
+	.limbsLength = {
+		-930.0f,
+		-1530.0f +  930.0f,
+		-2160.0f + 1530.0f,
+		-3150.0f + 2160.0f,
+		-4090.0f + 3150.0f,
+		-5190.0f + 4090.0f,
+		-6070.0f + 5190.0f,
+		-6810.0f + 6070.0f,
+		-400.0f,
+	},
+};
+```
+[Back to top](#list-of-content)
+
+---
+### ActorInit
+```c
+void EnNpc_Init(EnNpc* this, GlobalContext* globalCtx) {
+	Actor_SetScale(&this->actor, 0.0055f);
+	SkelAnime_InitFlex(globalCtx, &this->skelAnime, SKEL_NPC, ANIM_IDLE, this->jointTable, this->morphTable, SKEL_NPC_NUMBONES_DT);
+	
+	Physics_SetPhysicsStrand(&sPonytailInit, &this->phyPonytail, this->phyJointLength, this->bodyPartsPos);
+}
+```
+[Back to top](#list-of-content)
+
+---
+### OverridePostDraw
+In `SkelAnime` PostDraw you can get the "sphere" collision centers like so. This will prevent the limbs of the strand go through them.
+```c
+void EnNpc_PostDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, void* thisx) {
+	EnNpc* this = (EnNpc*)thisx;
+	Vec3f zero = { 0 };
+	
+	switch (limbIndex) {
+		case NpcLimb_Head:
+			Matrix_MultVec3f(&zero, &this->actor.focus.pos);
+			break;
+		case NpcLimb_Ponytail1:
+			Physics_GetHeadProperties(&this->phyPonytail, &zero, 0);
+			break;
+	    case NpcLimb_Upperbody:
+		    Matrix_MultVec3f(&zero, &this->bodyPartsPos[0]);
+		    break;
+	    case NpcLimb_Upperarm_R:
+		    Matrix_MultVec3f(&zero, &this->bodyPartsPos[1]);
+		    break;
+	    case NpcLimb_Thigh_R:
+		    Matrix_MultVec3f(&zero, &this->bodyPartsPos[2]);
+		    break;
+	    case NpcLimb_Waist:
+		    Matrix_MultVec3f(&zero, &this->bodyPartsPos[3]);
+		    break;
+	    case NpcLimb_Upperarm_L:
+		    Matrix_MultVec3f(&zero, &this->bodyPartsPos[4]);
+		    break;
+	    case NpcLimb_Thigh_L:
+		    Matrix_MultVec3f(&zero, &this->bodyPartsPos[5]);
+		    break;
+	}
+}
+```
+[Back to top](#list-of-content)
+
+---
+### PhysicCallback
+This callback can have up to two `pointers` of your choice. Type of the pointers are declared in this callback function itself and then assigned to the `Physics_DrawDynamicStrand` function if used.
+```c
+void EnNpc_PhysicsCallback(s32 limbIndex, GlobalContext* globalCtx, void* arg2) {
+	Vec3f zero = { 0.0f, 100.0f, 0.0f };
+
+	if (limbIndex == LIMB_END)
+		Matrix_MultVec3f(&zero, &this->tailEnd);
+}
+```
+[Back to top](#list-of-content)
+
+---
+### ActorDraw
+In draw you can do a `for loop` to get the positions calculated instantly. Also remember to use the extra arguments if you're going to use `PhysicCallback`.
```

Comment:
wouldn't it make more sense to move that loop to the init function

Reply:
Yeah it would, but it requires the head properties in order to get correct results. One way could be taking SkelAnime_DrawFlexOpa and using it in Init maybe.

Reply:
didn't think of that, yeah I guess moving to init would be more trouble than what it's worth. It's fine as is